crates/accelerate/src/synthesis/clifford/utils.rs

```diff
@@ -149,6 +149,30 @@ impl Clifford {
         azip!((z in &mut z, &x in &x) *z ^= x);
     }
 
+    /// Modifies the tableau in-place by appending SX-gate
+    pub fn append_sx(&mut self, qubit: usize) {
```

Comment:
In general, the Rust-based Clifford code is implicitly tested when checking the correctness of synthesis results in `test_high_level_synthesis.py`, etc.. However, this is not a full-proof method since this only considers gates produced by the synthesis algorithms. This is why we did not need handling `SX` and `SXsg` gates before this PR, none of our current Clifford synthesis algorithms produced such gates. 

The best way I can think of how to apply your suggestion is to add Rust-based tests that we would then execute with `cargo test`. However, so far we have been mostly avoiding this path. Note that our internal Rust-based Clifford code is not exposed via a Python interface, this is why I don't see how to write directed tests from the Python space. Would love to hear additional suggestions.